<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PasswordManagerClass</class>
 <widget class="QMainWindow" name="PasswordManagerClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>700</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>PasswordManager</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QFormLayout" name="formLayout">
    <item row="0" column="0" colspan="2">
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="SelectFile">
        <item>
         <widget class="QLabel" name="select_file_label">
          <property name="font">
           <font>
            <pointsize>11</pointsize>
           </font>
          </property>
          <property name="text">
           <string>选择文件：</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="file_path">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="font">
           <font>
            <pointsize>11</pointsize>
           </font>
          </property>
          <property name="echoMode">
           <enum>QLineEdit::EchoMode::Normal</enum>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="select_file">
          <property name="font">
           <font>
            <pointsize>11</pointsize>
           </font>
          </property>
          <property name="text">
           <string>打开</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="new_file">
          <property name="font">
           <font>
            <pointsize>11</pointsize>
           </font>
          </property>
          <property name="text">
           <string>新建</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="save_file">
          <property name="font">
           <font>
            <pointsize>11</pointsize>
           </font>
          </property>
          <property name="text">
           <string>保存</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="Password">
        <item>
         <widget class="QLabel" name="enter_password_label">
          <property name="font">
           <font>
            <pointsize>11</pointsize>
           </font>
          </property>
          <property name="text">
           <string>输入密码：</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="password_entry">
          <property name="font">
           <font>
            <pointsize>11</pointsize>
           </font>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="view_button">
          <property name="font">
           <font>
            <pointsize>11</pointsize>
           </font>
          </property>
          <property name="text">
           <string>显示</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="confirm_password">
          <property name="font">
           <font>
            <pointsize>11</pointsize>
           </font>
          </property>
          <property name="text">
           <string>确定</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="reset_password">
          <property name="font">
           <font>
            <pointsize>11</pointsize>
           </font>
          </property>
          <property name="text">
           <string>设置密码</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTableView" name="tableView">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="font">
         <font>
          <pointsize>11</pointsize>
         </font>
        </property>
        <property name="autoFillBackground">
         <bool>false</bool>
        </property>
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarPolicy::ScrollBarAsNeeded</enum>
        </property>
        <property name="editTriggers">
         <set>QAbstractItemView::EditTrigger::DoubleClicked</set>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::SelectionMode::SingleSelection</enum>
        </property>
        <property name="verticalScrollMode">
         <enum>QAbstractItemView::ScrollMode::ScrollPerPixel</enum>
        </property>
        <property name="horizontalScrollMode">
         <enum>QAbstractItemView::ScrollMode::ScrollPerPixel</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="Action">
        <item>
         <layout class="QHBoxLayout" name="Copy">
          <item>
           <widget class="QPushButton" name="copy_username">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>0</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>11</pointsize>
             </font>
            </property>
            <property name="text">
             <string>复制用户名</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="copy_password">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>0</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>11</pointsize>
             </font>
            </property>
            <property name="text">
             <string>复制密码</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>298</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="Edit">
          <item>
           <widget class="QPushButton" name="push_button">
            <property name="font">
             <font>
              <pointsize>11</pointsize>
             </font>
            </property>
            <property name="text">
             <string>添加</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="delete_button">
            <property name="font">
             <font>
              <pointsize>11</pointsize>
             </font>
            </property>
            <property name="text">
             <string>删除</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="statusTip">
    <string/>
   </property>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>700</width>
     <height>33</height>
    </rect>
   </property>
   <widget class="QMenu" name="file">
    <property name="title">
     <string>文件</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
   </widget>
   <addaction name="file"/>
  </widget>
  <action name="actionNew">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::DocumentNew"/>
   </property>
   <property name="text">
    <string>新建</string>
   </property>
   <property name="toolTip">
    <string>新建</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::DocumentOpen"/>
   </property>
   <property name="text">
    <string>打开</string>
   </property>
   <property name="toolTip">
    <string>打开</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::DocumentSave"/>
   </property>
   <property name="text">
    <string>保存</string>
   </property>
   <property name="toolTip">
    <string>保存</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::DocumentSaveAs"/>
   </property>
   <property name="text">
    <string>另存为</string>
   </property>
   <property name="toolTip">
    <string>另存为</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="PasswordManager.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>select_file</sender>
   <signal>clicked()</signal>
   <receiver>PasswordManagerClass</receiver>
   <slot>on_select_file()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>478</x>
     <y>57</y>
    </hint>
    <hint type="destinationlabel">
     <x>544</x>
     <y>34</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>new_file</sender>
   <signal>clicked()</signal>
   <receiver>PasswordManagerClass</receiver>
   <slot>on_new_file()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>570</x>
     <y>51</y>
    </hint>
    <hint type="destinationlabel">
     <x>636</x>
     <y>37</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>confirm_password</sender>
   <signal>clicked()</signal>
   <receiver>PasswordManagerClass</receiver>
   <slot>on_confirm_password()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>579</x>
     <y>93</y>
    </hint>
    <hint type="destinationlabel">
     <x>579</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>view_button</sender>
   <signal>pressed()</signal>
   <receiver>PasswordManagerClass</receiver>
   <slot>on_show_password()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>454</x>
     <y>82</y>
    </hint>
    <hint type="destinationlabel">
     <x>529</x>
     <y>76</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>view_button</sender>
   <signal>released()</signal>
   <receiver>PasswordManagerClass</receiver>
   <slot>on_hide_password()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>467</x>
     <y>98</y>
    </hint>
    <hint type="destinationlabel">
     <x>543</x>
     <y>76</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>push_button</sender>
   <signal>clicked()</signal>
   <receiver>PasswordManagerClass</receiver>
   <slot>on_push_new()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>551</x>
     <y>351</y>
    </hint>
    <hint type="destinationlabel">
     <x>466</x>
     <y>369</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>delete_button</sender>
   <signal>clicked()</signal>
   <receiver>PasswordManagerClass</receiver>
   <slot>on_remove_row()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>642</x>
     <y>361</y>
    </hint>
    <hint type="destinationlabel">
     <x>643</x>
     <y>371</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>menuBar</sender>
   <signal>triggered(QAction*)</signal>
   <receiver>PasswordManagerClass</receiver>
   <slot>on_menu(QAction*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>21</x>
     <y>12</y>
    </hint>
    <hint type="destinationlabel">
     <x>25</x>
     <y>39</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>save_file</sender>
   <signal>clicked()</signal>
   <receiver>PasswordManagerClass</receiver>
   <slot>on_save_file()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>678</x>
     <y>56</y>
    </hint>
    <hint type="destinationlabel">
     <x>690</x>
     <y>78</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>reset_password</sender>
   <signal>clicked()</signal>
   <receiver>PasswordManagerClass</receiver>
   <slot>on_set_password()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>677</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>697</x>
     <y>107</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tableView</sender>
   <signal>activated(QModelIndex)</signal>
   <receiver>PasswordManagerClass</receiver>
   <slot>on_view(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>240</x>
     <y>237</y>
    </hint>
    <hint type="destinationlabel">
     <x>240</x>
     <y>343</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tableView</sender>
   <signal>pressed(QModelIndex)</signal>
   <receiver>PasswordManagerClass</receiver>
   <slot>on_view(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>312</x>
     <y>304</y>
    </hint>
    <hint type="destinationlabel">
     <x>311</x>
     <y>356</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tableView</sender>
   <signal>doubleClicked(QModelIndex)</signal>
   <receiver>PasswordManagerClass</receiver>
   <slot>on_edit(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>402</x>
     <y>313</y>
    </hint>
    <hint type="destinationlabel">
     <x>405</x>
     <y>341</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>copy_username</sender>
   <signal>clicked()</signal>
   <receiver>PasswordManagerClass</receiver>
   <slot>on_copy_username()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>67</x>
     <y>353</y>
    </hint>
    <hint type="destinationlabel">
     <x>63</x>
     <y>371</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>copy_password</sender>
   <signal>clicked()</signal>
   <receiver>PasswordManagerClass</receiver>
   <slot>on_copy_password()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>154</x>
     <y>353</y>
    </hint>
    <hint type="destinationlabel">
     <x>158</x>
     <y>373</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>password_entry</sender>
   <signal>returnPressed()</signal>
   <receiver>PasswordManagerClass</receiver>
   <slot>on_confirm_password()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>246</x>
     <y>96</y>
    </hint>
    <hint type="destinationlabel">
     <x>252</x>
     <y>109</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>on_select_file()</slot>
  <slot>on_new_file()</slot>
  <slot>on_confirm_password()</slot>
  <slot>on_show_password()</slot>
  <slot>on_hide_password()</slot>
  <slot>on_add_data(QString,QString,QString)</slot>
  <slot>on_push_new()</slot>
  <slot>on_remove_row()</slot>
  <slot>on_menu(QAction*)</slot>
  <slot>on_save_file()</slot>
  <slot>on_set_password()</slot>
  <slot>on_view(QModelIndex)</slot>
  <slot>on_edit(QModelIndex)</slot>
  <slot>on_copy_username()</slot>
  <slot>on_copy_password()</slot>
 </slots>
</ui>
